lambda_2/lambda_function.py

```diff
@@ -0,0 +1,58 @@
+import json
+import csv
+import boto3
+import os
+from datetime import datetime
+
+def handler(event, context):
+    source_bucket_name = os.environ['BUCKET_NAME']
+    file_key = event['Records'][0]['s3']['object']['key']
+
+    output_file_name = 'flattened_records.csv'
+
+    s3 = boto3.resource('s3')
+    source_bucket = s3.Bucket(source_bucket_name)
+
+    json_file = source_bucket.Object(file_key).get()['Body'].read().decode('utf-8')
+    data = json.loads(json_file)
+
+    flattened_data = process_json(data)
+
+    with open('/tmp/flattened_records.csv', 'w', newline='') as outfile:
```

Comment:
An alternate, quicker way here could be to read the flattened records list into a dataframe, then using wr.s3.to_csv() to avoid needing the csv writer. Up to you if you want to change it though :)

Reply:
got it.